### ISSUE TITLE
Build error on master on Windows 10

### ISSUE THREAD

**Issue by ChrisRackauckas** [dmlc/XGBoost.jl#51]
After checking out master, I tried to build:

```julia

Pkg.build("XGBoost")

INFO: Building XGBoost
INFO: Using the latest stable version (0.60) of the XGBoost library
===============================[ ERROR: XGBoost ]===============================

LoadError: None of the selected providers can install dependency xgboost.
Use BinDeps.debug(package_name) to see available providers

while loading C:\Users\Chris\.julia\v0.6\XGBoost\deps\build.jl, in expression starting on line 36

================================================================================

================================[ BUILD ERRORS ]================================

WARNING: XGBoost had build errors.

 - packages with build errors remain installed in C:\Users\Chris\.julia\v0.6
 - build the package(s) and all dependencies with `Pkg.build("XGBoost")`
 - build a single package by running its `deps/build.jl` script

================================================================================
```

**Comment by oxinabox** [comment]
Just some notes:

 - backend does support window.

 - The current `deps/build.jl` uses commandline calls to `git clone` and build the back end.
 - Rather that the more common options of downloading a precompiled binary, or using a system package manager (Apt+Yum+WinRPM, or Conda)
 - Conda Forge does not windows https://github.com/conda-forge/xgboost-feedstock/issues/3
- default/andaconda channel does have it though: https://anaconda.org/anaconda/libxgboost

**Comment by ChrisRackauckas** [comment]
I downloaded the binary from 

http://www.picnet.com.au/blogs/guido/post/2016/09/22/xgboost-windows-x64-binaries-for-download/

and then I setup a deps.jl that was:

```julia
# This is an auto-generated file; do not edit

# Pre-hooks

# Macro to load a library
macro checked_lib(libname, path)
    ((VERSION >= v"0.4.0-dev+3844" ? Base.Libdl.dlopen_e : Base.dlopen_e)(path) == C_NULL) && error("Unable to load \n\n$libname ($path)\n\nPlease re-run Pkg.build(package), and restart Julia.")
    quote const $(esc(libname)) = $path end
end

# Load dependencies
@checked_lib _xgboost "C:/Users/Chris/.julia/v0.6/XGBoost/deps/bin/xgboost.dll"
@checked_lib _jl_libxgboost "C:/Users/Chris/.julia/v0.6/XGBoost/deps/bin/xgboost.dll"

# Load-hooks
global const build_version = "master"
```

and then it worked. It throws 

```julia
[1]	train-error:0.024200
[2]	train-error:0.013900
[3]	train-error:0.006900
[4]	train-error:0.003400
[5]	train-error:0.001900
[6]	train-error:0.000300
[7]	train-error:0.000200
[8]	train-error:0.000000
[9]	train-error:0.000000
```

every step, but the output is correct.

**Comment by oxinabox** [comment]
`((VERSION >= v"0.4.0-dev+3844" ? Base.Libdl.dlopen_e : Base.dlopen_e)(path) == C_NULL) && error("Unable to load \n\n$libname ($path)\n\nPlease re-run Pkg.build(package), and restart Julia.")`

can be:

`(Base.Libdl.dlopen_e == C_NULL) && error("Unable to load \n\n$libname ($path)\n\nPlease re-run Pkg.build(package), and restart Julia.")`

Since XGBoost requires Julia 0.5

**Comment by aviks** [comment]
Fixed by #68
